pkg/virt-launcher/virtwrap/device/sriov/hostdev_test.go

```diff
@@ -280,6 +280,43 @@ var _ = Describe("SRIOV HostDevice", func() {
 			Expect(sriov.SafelyDetachHostDevices(domainSpec, c, d, 1)).To(Succeed())
 		})
 	})
+
+	Context("attachment", func() {
+		hostDevice := api.HostDevice{Alias: api.NewUserDefinedAlias("net1")}
+
+		It("ignores nil list of devices", func() {
+			Expect(sriov.AttachHostDevices(deviceAttacherFlagsStub{}, nil)).Should(Succeed())
+		})
+
+		It("ignores an empty list of devices", func() {
+			Expect(sriov.AttachHostDevices(deviceAttacherFlagsStub{}, []api.HostDevice{})).Should(Succeed())
+		})
+
+		It("succeeds to attach device", func() {
+			Expect(sriov.AttachHostDevices(deviceAttacherFlagsStub{}, []api.HostDevice{hostDevice})).Should(Succeed())
+		})
+
+		It("succeeds to attach more than one device", func() {
+			hostDevice2 := api.HostDevice{Alias: api.NewUserDefinedAlias("net2")}
+
+			Expect(sriov.AttachHostDevices(deviceAttacherFlagsStub{}, []api.HostDevice{hostDevice, hostDevice2})).Should(Succeed())
+		})
+
+		It("fails to attach device", func() {
+			obj := deviceAttacherFlagsStub{fail: true}
+			Expect(sriov.AttachHostDevices(obj, []api.HostDevice{hostDevice})).ShouldNot(Succeed())
+		})
+
+		It("error should contain at least that Alias of each device that failed to attach", func() {
+			obj := deviceAttacherFlagsStub{fail: true}
+			hostDevice2 := api.HostDevice{Alias: api.NewUserDefinedAlias("net2")}
+			err := sriov.AttachHostDevices(obj, []api.HostDevice{hostDevice,hostDevice2})
+			Expect(err).ToNot(BeNil())
```

Comment:
You can wrap `sriov.AttachHostDevices` and use `Succeed`.

Reply:
I reuse the `err` for the assertion on line 315

Reply:
The convention is then to use `HaveOccurred()`.